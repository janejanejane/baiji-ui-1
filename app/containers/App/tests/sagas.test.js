import { put } from 'redux-saga/effects';

import {
  meRequestSuccess,
  meRequestFailed,
} from '../actions';

import {
  getMeFromToken,
} from '../sagas';

describe('Request Me From Token Saga', () => {
  let getMeFromTokenGenerator;

  beforeEach(() => {
    global.sessionStorage = jest.fn();
    global.sessionStorage.setItem = jest.fn();
    global.sessionStorage.removeItem = jest.fn();

    getMeFromTokenGenerator = getMeFromToken({
      token: 'exampletoken',
    });
    const testCall = getMeFromTokenGenerator.next().value;

    expect(testCall).toMatchSnapshot();
  });

  describe('OnSuccess', () => {
    it('should dispatch meRequestSuccess on success', () => {
      const fixture = {
        username: 'testusername',
      };

      const callDescriptor = getMeFromTokenGenerator.next(fixture).value;
      expect(callDescriptor).toEqual(put(meRequestSuccess(fixture)));
    });

    it('should set jwtToken in sessionStorage', () => {
      const fixture = {
        username: 'testusername',
        token: 'token',
      };

      getMeFromTokenGenerator.next(fixture);
      expect(global.sessionStorage.setItem).toHaveBeenCalledWith('jwtToken', fixture.token);
    });
  });

  describe('OnFail', () => {
    it('should dispatch meRequestFailed on failure', () => {
      const fixture = {
        message: 'message',
      };

      const callDescriptor = getMeFromTokenGenerator.throw(fixture).value;
      expect(callDescriptor).toEqual(put(meRequestFailed(fixture.message)));
    });

    it('should remove jwtToken', () => {
      const fixture = {
        message: 'message',
      };

      getMeFromTokenGenerator.throw(fixture);
      expect(global.sessionStorage.removeItem).toHaveBeenCalledWith('jwtToken');
    });
  });
});
