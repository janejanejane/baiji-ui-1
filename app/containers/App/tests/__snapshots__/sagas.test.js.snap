// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Me From Token Saga should dispatch meRequestFailed on failure 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/me?token=exampletoken",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Request Me From Token Saga should dispatch meRequestSuccess on success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/me?token=exampletoken",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
